Add report on outbreak distribution using a branching process (#2)
jamesmbaazam/h5n1_uk_scenario_modelling: report/outbreak_no_intervention.qmd

---
title: H5N1 outbreak scenarios
---

> This report requires two R packages: {epichains} and {ggplot2}, the former to simulate the branching process model and the latter to visualise the results.

Reviewer: I'm not sure we need this here. The scientific report should not have any code or code-related references. We could add it under a section on "code and data availability" or something of the sort.

```suggestion

```

> ```{r}

Reviewer: ```suggestion
```{r echo=FALSE}
```

# install from outbreak_dist branch of {epichains}
# remotes::install_github("epiverse-trace/epichains@outbreak_dist")
library(epichains)
library(ggplot2)
```

> The `simulate_scenarios()` function from {epichains} calculates the proportion of outbreak statistic across values of the reproduction number ($R$).

Reviewer: Similar to my rpevious comment, I'm not sure this detail should be in this report as we are reporting the science and not the engineering, so I would suggest to move it to an appendix or section on "code and data".
```suggestion
The `simulate_scenarios()` function from {epichains} calculates the proportion of outbreak statistic (size and/or length) across scenarios of reproduction number ($R0$) values, overdispersion, $k$, and offspring distributions.
```

> Here we run the function for both Poisson (`"rpois"`) and Negative Binomial (`"rnbinom"`) distributions, calculating both the size and length of the outbreak. Size is the total number of cases produced by a chain before it goes extinct. Length is the total number of generations reached by a chain before it goes extinct. See `?epichains::simulate_chain_stats` for more information on the branching process simulation.
> 
> We simulate using a reproduction number ($R$) between 0.1 and 0.9 at 0.1 intervals, and for the Negative Binomial offspring distribution we vary the dispersion parameter ($k$) between 0.5 and 2.5 at 0.5 intervals. Smaller values of $k$ signify more overdispersion in the transmission spreading, in other words transmission is more heterogeneous and superspreding events are more likely. As $k$ increases individual-level transmission is more homogeneous and when $k \rightarrow \infty$ the Negative Binomial equals the Poisson offspring distribution. 

Reviewer: ```suggestion
Here, we use a simple branching process to explore scenarios of reproduction number ($R0$) values, overdispersion, $k$, and offspring distributions and their associated chain size and length. The branching process is seeded with an initial number of cases and an offspring distribution, and optionally, a generation time distribution.

We define the chain size as the total number of cases produced by a chain before it goes extinct. The chain length, on the other hand, is the total number of generations reached by a chain before it goes extinct.

We use a reproduction number ($R$) between 0.1 and 0.9 at 0.1 intervals, and for the Negative Binomial offspring distribution, we vary the dispersion parameter ($k$) between 0.5 and 2.5 at 0.5 intervals. Smaller values of $k$ signify more overdispersion in the individual-level transmissibility, in other words transmission is more heterogeneous and superspreading events are more likely. As $k$ increases individual-level transmission is more homogeneous and when $k \rightarrow \infty$ the Negative Binomial equals the Poisson offspring distribution. 
```

```{r}
df <- simulate_scenarios(
  offspring_dist = c("rpois", "rnbinom"), 
  statistic = c("size", "length"), 
  R_seq = seq(0, 0.9, 0.1), 
  k_seq = seq(0.5, 2.5, 0.5), 
  breaks = c(0, 2, 5, 10, 20, 50, 100, Inf)
)
```

> The function produces a `data.frame` that contains the outbreak size or length in intervals (`$interval`), the proportion of outbreaks that fall into that interval (`$Freq`), the reproduction number (`$R`), the dispersion parameter (`$k`, for Poisson distributions $k$ is set to `NA`), the offspring distribution (`$offspring_dist`), and the outbreak statistic (`$statistic`, either size or length). 

Reviewer: Would suggest to cut as well.

```suggestion
The function produces a `data.frame` that contains the outbreak size or length in intervals (`$interval`), the proportion of outbreaks that fall into that interval (`$Freq`), the reproduction number (`$R`), the dispersion parameter (`$k`, for Poisson distributions $k$ is set to `NA`), the offspring distribution (`$offspring_dist`), and the outbreak statistic (`$statistic`, either size or length). 
```

Author: I'm going to leave this in for now as the report is still WIP. We can easily remove it if we agree it can go in the future.

> Here are the first 6 rows.

Reviewer: ```suggestion

```

Author: Same as above, I'm not sure what this report will be used for and these things are easy to remove in the future, so for now I'll leave them in for clarity.

> ```{r}
> head(df)
> ```

Reviewer: ```suggestion

```

Author: Same reasoning as above.

> Plotting the Poisson results.

Reviewer: ```suggestion
Below, we show the results for the Poisson distribution for varying plausible $R0$ values. The figure shows the proportion of outbreaks that achieved different outbreak size ranges.
```

```{r}
#| code-fold: true
#| code-summary: "Show the code"
df |> 
  subset(offspring_dist == "rpois") |>
  ggplot2::ggplot() +
  ggplot2::geom_col(
    mapping = ggplot2::aes(x = as.factor(R), y = Freq, fill = interval)
  ) +
  ggplot2::scale_x_discrete(name = "Reproduction number (R)") +
  ggplot2::scale_y_continuous(name = "Proportion of outbreaks") +
  ggplot2::scale_fill_brewer(name = "Outbreak size", palette = "Spectral") +
  ggplot2::facet_wrap(
    facets = c("offspring_dist", "statistic"), 
    labeller = ggplot2::label_both
  ) +
  ggplot2::labs(
    title = "Outbreak size and length for Poisson Branching Process"
  ) +
  ggplot2::theme_bw() + 
  ggplot2::theme(strip.background = ggplot2::element_blank())
```

> Plotting the Negative Binomial results for $k = 0.5$.

Reviewer: ```suggestion
Next are the results of the negative distribution offspring for varying plausible $R0$ values but a fixed $k$ of $0.5$. The figure shows the proportion of outbreaks that achieved different outbreak size ranges.
```

```{r}
#| code-fold: true
#| code-summary: "Show the code"
df |> 
  subset(offspring_dist == "rnbinom" & k == 0.5) |>
  ggplot2::ggplot() +
  ggplot2::geom_col(
    mapping = ggplot2::aes(x = as.factor(R), y = Freq, fill = interval)
  ) +
  ggplot2::scale_x_discrete(name = "Reproduction number (R)") +
  ggplot2::scale_y_continuous(name = "Proportion of outbreaks") +
  ggplot2::scale_fill_brewer(name = "Outbreak size", palette = "Spectral") +
  ggplot2::facet_wrap(
    facets = c("offspring_dist", "statistic"), 
    labeller = ggplot2::label_both
  ) +
  ggplot2::labs(
    title = "Outbreak size and length for Negative Binomial Branching Process"
  ) +
  ggplot2::theme_bw() + 
  ggplot2::theme(strip.background = ggplot2::element_blank())
```

> Plotting the outbreak size distribution for the Negative Binomial results faceting for values of $k$.

Reviewer: ```suggestion
We now show the results of the outbreak using a negative distribution offspring for varying plausible values of $R0$ and $k$. The figure shows the proportion of outbreaks that achieved different outbreak size ranges.
```

```{r}
#| code-fold: true
#| code-summary: "Show the code"
df |> 
  subset(offspring_dist == "rnbinom" & statistic == "size") |>
  ggplot2::ggplot() +
  ggplot2::geom_col(
    mapping = ggplot2::aes(x = as.factor(R), y = Freq, fill = interval)
  ) +
  ggplot2::scale_x_discrete(name = "Reproduction number (R)") +
  ggplot2::scale_y_continuous(name = "Proportion of outbreaks") +
  ggplot2::scale_fill_brewer(name = "Outbreak size", palette = "Spectral") +
  ggplot2::facet_wrap(
    facets = c("k"), 
    labeller = ggplot2::label_both
  ) +
  ggplot2::labs(
    title = "Outbreak size Negative Binomial Branching Process"
  ) +
  ggplot2::theme_bw() + 
  ggplot2::theme(strip.background = ggplot2::element_blank())
```

> Plotting the outbreak length distribution for the Negative Binomial results faceting for values of $k$.

Reviewer: ```suggestion
In the next figure, we show the results of outbreak lengths for a negative binomial offspring distribution with varying $R0$ and $k$ values.
```
